File: acme-java-springboot/src/main/java/acme/entity/Bank.java
package acme.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Version;
import org.springframework.data.relational.core.mapping.Table;

import java.util.UUID;

@Table("bank")
public record Bank(
    @Id UUID bankId,
    String name,
    String address,
    @Version Long version
) {
    public Bank(UUID bankId, String name, String address) {
        this(bankId, name, address, 0L);
    }
}

File: acme-springevent/src/main/kotlin/acme/entity/Bank.kt
package acme.entity

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Version
import org.springframework.data.relational.core.mapping.Table
import java.util.*

@Table("bank")
data class Bank(
    @Id
    val bankId: UUID,
    val name: String,
    val address: String,
    @Version
    val version: Long = 0
)
